test/library/assertions.go

```diff
@@ -24,3 +28,63 @@ func RequireEventuallyWithoutError(
 	t.Helper()
 	require.NoError(t, wait.PollImmediate(tick, waitFor, f), msgAndArgs...)
 }
+
+// NewRestartAssertion allows a caller to assert that there were no restarts for a Pod in the
+// provided namespace with the provided labelSelector.
+//
+// It returns a func that should be called at the end of a test.
+//
+//   func TestSomething(t *testing.T) {
+//     restartAssertion := library.NewRestartAssertion(t, "my-namespace", "app=my-app")
+//     defer restartAssertion()
+//     // Rest of the test...
+//   }
+func NewRestartAssertion(t *testing.T, namespace, labelSelector string) func() {
```

Comment:
I think this is fine but there is one other way we could write this using `t.Cleanup()` that might be a little simpler:
```go
func AssertNoRestartsDuringTest(t *testing.T, namespace, labelSelector string) {
   // capture initial counts
  t.Cleanup(func(){
    // capture counts after test
    // assertions...
  })

}
```

The usage then becomes a single line like:
```go
library.AssertNoRestartsDuringTest(t, "my-namespace", "app=my-app")
```

Reply:
Dang. That is easier. I'm not sure why I didn't do that...oh, I think I tried to stick to the pattern of `defer library.DumpLogs()` since I read that in the description of the issue, but I like this `t.Cleanup()`-based implementation much better. At some point I will update this PR to use that implementation.

Reply:
Gave this a go in https://github.com/vmware-tanzu/pinniped/pull/325/commits/42ff247a54939a9b80538af71c9bc6f572ea1559 whenever you have a couple minutes to take a look.